server/gateway.go

```diff
@@ -1122,8 +1144,9 @@ func (c *client) processGatewayInfo(info *Info) {
 		js := s.js
 		s.mu.Unlock()
 
-		// Switch JetStream accounts to interest-only mode.
-		if js != nil {
+		// If running in some tests, maintain the original behavior.
+		if gwDoNotForceInterestOnlyMode && js != nil {
```

Comment:
Should we still do this if the server not running JS but has accounts with JS enabled with limits?

Reply:
This is only for tests that want to maintain the old behavior. If not for that test variable (say the executable code compiled outside of test framework), the "else" (without even the check for the variable since it will be always false so compiled out) clause is executed, which is to switch ALL accounts.